### PR TITLE
Introduce `collectives_polkadot_runtime_constants` and removed hard-coded numbers usage for indexes

### PR DESCRIPTION
**Pull request by bkontur**
This PR solves older TODO + addresses https://github.com/polkadot-fellows/runtimes/pull/109#discussion_r1420818182.

- [X] Does not require a CHANGELOG entry

### CODE REVIEW

File: system-parachains/collectives/collectives-polkadot/constants/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
```
Comment (acatangiu) [inline system-parachains/collectives/collectives-polkadot/constants/src/lib.rs:17]: can we drop the `std` feature altogether for this crate?

**Review by acatangiu — APPROVED** [review]

File: system-parachains/collectives/collectives-polkadot/constants/src/lib.rs (hunk 1 above)
Comment (bkontur) [reply system-parachains/collectives/collectives-polkadot/constants/src/lib.rs:17]: Do you mean also to remove from Cargo.toml:
```
[features]
default = ["std"]
std = []
```
and also usage:
```
std = [
    ...
    "collectives-polkadot-runtime-constants/std",
    ...
```
?

Actually, I don't know, I don't see any/much libraries used in the runtimes without this stuff. I don't know, can we remove it?

Comment (acatangiu) [reply system-parachains/collectives/collectives-polkadot/constants/src/lib.rs:17]: afaict this is only used by runtime code which is compiled to wasm always without `std`, so it should work

Comment (bkontur) [reply system-parachains/collectives/collectives-polkadot/constants/src/lib.rs]: removed, yes, looks like everything is ok (at least locally), lets wait for the CI

Comment (bkontur) [reply system-parachains/collectives/collectives-polkadot/constants/src/lib.rs]: hmm, strange, CI complains: https://github.com/polkadot-fellows/runtimes/actions/runs/8111248007/job/22170156363?pr=182

Comment (bkontur) [reply system-parachains/collectives/collectives-polkadot/constants/src/lib.rs:17]: after `cargo clean, it does not work locally also, I had to revert it

Comment (acatangiu) [reply system-parachains/collectives/collectives-polkadot/constants/src/lib.rs:17]: ok, at least we tried

**Review by acatangiu — APPROVED** [review]

File: system-parachains/collectives/collectives-polkadot/constants/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+name = "collectives-polkadot-runtime-constants"
+repository.workspace = true
+version.workspace = true
+authors.workspace = true
+edition.workspace = true
+license.workspace = true
+
+[features]
+default = ["std"]
+std = []
```
Comment (bkchr) [inline system-parachains/collectives/collectives-polkadot/constants/Cargo.toml:11]:
```suggestion
```

**Pull request merged**